### ISSUE TITLE
VEP Docker and converted cache

### ISSUE THREAD

**Issue by sigven** [Ensembl/ensembl-vep#377]
Hi,

I just pulled the latest Docker image [https://hub.docker.com/r/ensemblorg/ensembl-vep/tags]( https://hub.docker.com/r/ensemblorg/ensembl-vep/tags). 

I also downloaded the latest GRCh37 cache: [ftp://ftp.ensembl.org/pub/current_variation/VEP/homo_sapiens_vep_95_GRCh37.tar.gz](ftp://ftp.ensembl.org/pub/current_variation/VEP/homo_sapiens_vep_95_GRCh37.tar.gz), which I subsequently converted with tabix using

`convert_cache.pl --species homo_sapiens --version 95_GRCh37 --remove`

Next, I made a simple annotation test with an [example VCF](https://drive.google.com/open?id=1f-iIYzbOkYh5NecQAblE_D5KfcE_HnIT) with some ~9,000 variants that I have tested successfully with previous Docker releases. Now, the run will not complete, it halts at the following warning:

`Warning: The index file is older than the data file: /usr/local/share/vep/data/homo_sapiens/95_GRCh37/1/all_vars.gz.tbi`

I am unsure whether this is an issue with my Mac Docker setup (2.0.0.0-mac81, engine currently set up with memory of 8Gb and 4CPUs), or if it is related to the converted cache(?), or something else. I did not use any forking, and tried also to lower the buffer_size, but that did not help.

Any feedback appreciated,

Thanks,
Sigve

**Comment by sigven** [comment]
Just tested the same VCF with ensembl-vep version 94.5 (and tabix-converted cache 94_GRCh37), and that worked fine within the Docker environment.

**Comment by ens-lgil** [comment]
Dear @sigven,

It might be because the `all_vars.gz` and its index file `all_vars.gz.tbi` have created close together in time. However this is just a warning message.
An other solution (that we are going to promote more in the incoming releases) is to download a version of the cache already indexed:
[ftp://ftp.ensembl.org/pub/current_variation/indexed_vep_cache/homo_sapiens_vep_95_GRCh37.tar.gz](ftp://ftp.ensembl.org/pub/current_variation/indexed_vep_cache/homo_sapiens_vep_95_GRCh37.tar.gz).

Best regards,
Laurent

**Comment by sigven** [comment]
Ok thanks, that will be great! I was actually wondering if you could release those pre-indexed caches (cause the tabix conversion takes quite some time..:)). I'll have a go with the pre-indexed cache, but as you say, that was merely a warning, and not sure if that may resolve my why the annotations halts.

**Comment by sigven** [comment]
Docker works now with build 95 using your pre-indexed cache:-) Thanks a lot!

**Comment by ens-lgil** [comment]
Dear @sigven,

This is expected as we imported dbSNP 151 for Human on GRCh37 in the release 95 and this version has much more variants compared to the previous versions of dbSNP we had in Ensembl 94 for GRCh37.
This increase is likely to have an impact on the overall speed, however the impact should be quite limited with the cache and the indexed cache.
Furthermore we are planning to switch the VEP Docker base from Ubuntu 16.04 to Ubuntu 18.04 next release and we noticed that the computing of VEP is slighlty quicker with Ubuntu 18.04.

Best regards,
Laurent
